wrappers/node/test/suite1/ariesvcx-proof.test.ts

```diff
@@ -69,9 +71,18 @@ describe('Proof:', () => {
   describe('deserialize:', () => {
     it('success', async () => {
       const proof1 = await proofCreate()
+      console.log(`Original proof1 ${JSON.stringify(proof1)}`)
       const data1 = await proof1.serialize()
+      // console.log(`Serialized proof1 = ${JSON.stringify(data1)}`)
       const proof2 = await Proof.deserialize(data1)
-      assert.equal(proof2.sourceId, proof1.sourceId)
+      // console.log(`Deserialized proof1data = ${JSON.stringify(proof2)}`)
+      // todo: Does not hold in aries, the TS/JS representation after serialize->deserialize in incorrect because
+      // IProofData structure is matching legacy structure
+      // perhaps we could make JS layer thinner and instead of trying to keeping attributes like _requestedAttributes
+      // in javascript representations, we could rather add method on libvcx proof vcx_proof_get_requested_attributes
+      // which could encompass the logic of how to retrieve this data from internal proof representation.
+      // The downside is some overhead associated with FFI.
```

Comment:
Honestly, I don't like the idea of adding `vcx_proof_get_requested_attributes`, requested attributes should be part of the serialized proof object in the first place. And since the wrapper is written in TS, this forces us to type the expected structure - and the benefits are arguable...

Anyway, no need to keep the this part of the comment here.

Reply:
They surely are part of serialized proof object. The question is where is the data stored once it's deserialized - rust layer or nodejs layer. 

I'd like to keep the comment because it's something to resolve in the future / next steps.

Reply:
Why do we need a new method? Can't we get the requested attributes from Rust layer by serialization?

Comments are not the place to keep things to resolve in the future, we have github issues for this.